frontends/concrete-python/concrete/fhe/mlir/context.py

```diff
@@ -2787,6 +2787,44 @@ def ones(self, resulting_type: ConversionType) -> Conversion:
 
         return self.add(resulting_type, one, self.zeros(resulting_type))
 
+    def relu(self, resulting_type: ConversionType, x: Conversion) -> Conversion:
+        if x.is_unsigned:
+            if resulting_type.bit_width == x.bit_width:
+                return x
+
+            assert resulting_type.bit_width > x.bit_width
+            return self.extract_bits(resulting_type, x, bits=slice(0, x.original_bit_width))
+
+        if x.original_bit_width == 1:
+            return self.zeros(resulting_type)
+
+        chunk_size = 2
+        intermediate_type = self.tensor(self.eint(chunk_size + 1), shape=x.shape)
+        sign = self.extract_bits(intermediate_type, x, bits=(x.original_bit_width - 1))
```

Comment:
you can extract the sign as 1 bit, and reinterpret as chunk_size + 1, this prevent from having to multiply the other extracted bits.